packages/credentials/src/status-list-credential.ts

```diff
@@ -0,0 +1,238 @@
+export const DEFAULT_STATUS_LIST_VC_TYPE = 'StatusList2021Credential';
+
+export enum StatusPurpose {
+  REVOCATION = 'revocation',
+  SUSPENSION = 'suspension',
+}
+
+/**
+ * The size of the bitstring in bits.
+ * The bitstring is 16KB in size.
+ */
+const BITSTRING_SIZE = 16 * 1024 * 8; // 16KB in bits
+
+/**
+ * StatusListCredentialCreateOptions for creating a status list credential.
+ *
+ * @param statusListCredentialId The id used for the resolvable path to the status list credential [String].
+ * @param issuer The issuer URI of the credential, as a [String].
+ * @param statusPurpose The status purpose of the status list cred, eg: revocation, as a [StatusPurpose].
+ * @param issuedCredentials The credentials to be included in the status list credential, eg: revoked credentials, list of type [VerifiableCredential].
+ */
+export type StatusListCredentialCreateOptions = {
+  statusListCredentialId: string,
+  issuer: string,
+  statusPurpose: StatusPurpose,
+  issuedCredentials: VerifiableCredential[]
+};
+
+/**
+ * The StatusList2021Entry instance representing the core data model of a vc status list 2021.
+ *
+ * @see {@link https://www.w3.org/community/reports/credentials/CG-FINAL-vc-status-list-2021-20230102/ | Status List 2021 Entry}
+ */
+export interface StatusList2021Entry extends ICredentialStatus {
+  statusListIndex: string,
+  statusListCredential: string,
+  statusPurpose: string,
+}
+
+/**
+ * `StatusListCredential` represents a digitally verifiable status list credential according to the
+ * [W3C Verifiable Credentials Status List v2021](https://www.w3.org/community/reports/credentials/CG-FINAL-vc-status-list-2021-20230102/).
+ *
+ * When a status list is published, the result is a verifiable credential that encapsulates the status list.
+ *
+ */
+export class StatusListCredential {
+  /**
+   * Create a [StatusListCredential] with a specific purpose, e.g., for revocation.
+   *
+   * @param statusListCredentialId The id used for the resolvable path to the status list credential [String].
+   * @param issuer The issuer URI of the credential, as a [String].
+   * @param statusPurpose The status purpose of the status list cred, eg: revocation, as a [StatusPurpose].
+   * @param issuedCredentials The credentials to be included in the status list credential, eg: revoked credentials, list of type [VerifiableCredential].
+   * @returns A special [VerifiableCredential] instance that is a StatusListCredential.
+   * @throws Error If the status list credential cannot be created.
+   *
+   * Example:
+   * ```
+      StatusListCredential.create({
+        statusListCredentialId : 'https://statuslistcred.com/123',
+        issuer                 : issuerDid.uri,
+        statusPurpose          : StatusPurpose.REVOCATION,
+        issuedCredentials      : [credWithCredStatus]
+      })
+   * ```
+   */
+  public static create(options: StatusListCredentialCreateOptions): VerifiableCredential {
```

Comment:
Design: from status list lifecycle management and dev experience standpoint, it would be more useful if:

1. There is an option that takes in an existing status list to be operated on.
    ie. Revoke a VC from an existing status list. This would be the main line scenario of status list usage.
2. It would be nice to support toggle from 1 to 0 also.
    ie. Reactivate a suspension of VC.

Reply:
Yup! You are right that is probably true the api needs to be updated

The current api copies that of the kt and the old go impl, The status list -is- regenerated every time there is an update, so its basically creating a new one I guess was the thought here.

I agree the api needs to be updated to be easier to grok. I'll create an iussue